webpack.common.js

```diff
@@ -97,6 +97,9 @@ const config = [
 			'data-stores': [
 				assets + 'data/index.js',
 			],
+			'eventespresso-core-blocks': [
```

Comment:
My understanding is that core-blocks will pick up everything that ultimately gets imported after loading `./assets/src/blocks/index.js` and there is no way for us to change the distributed JS files depending on the current route (ie: we will not be dynamically building those assets based on the route, right?), so we basically have no choice but to include the JS for every block in the bundle. But if we are building assets based on the specific route, then where does that happen and how?

I have repeatedly said that the logic for controlling the specific routes that blocks load on is not complete and will be done later on, and I assumed that that would not affect the distributed assets but only whether or not the block appears in the editor. IMHO it will be easier to attempt tackling route control after having some actual blocks up and running.

Reply:
My main concern with blocks is that they are ultimately registered client side which makes them show up in the GB interface.  So if they are all in a single bundled file, every GB editor instance will expose them for use regardless of any server side routing configuration.  That means we have to look more closely at _how_ we end up bundling as well.  One idea I had is currently implemented with the `wp-plugins-page` bundle.  I'm wondering if we could have certain block bundles built for the route they appear on and those would include all the blocks known for registration on that route. So for instance there might be a `wp-post-editor` bundle or a `wp-page-editor` bundle etc. I know the idea still needs fleshed out/discussed and I don't expect us to do as a part of this pull but my comment was merely to highlight it as an issue we'll have to tackle.  I totally agree with you made:

> IMHO it will be easier to attempt tackling route control after having some actual blocks up and running.

So I'm mostly just keen on keeping this issue on our minds.  I've created an issue to track this that we can use for planning a solution: https://github.com/eventespresso/event-espresso-core/issues/485